droidlet/lowlevel/hello_robot/remote/lidar.py

```diff
@@ -0,0 +1,35 @@
+from rplidar import RPLidar
+import sys
+import threading
+import time
+import traceback
+
+
+class Lidar:
+    def __init__(self):
+        self._lidar = RPLidar('/dev/hello-lrf')
+        self._lock = threading.Lock()
+        self._thread = None
+        self.latest_scan = None
+
+    def start(self):
+        self._thread = threading.Thread(target=self.lidar_loop, daemon=True)
+        self._thread.start()
+
+    def get_latest_scan(self):
+        with self._lock:
+            return self.latest_scan
+
+    def lidar_loop(self):
+        while True:
+            try:
+                for scan in self._lidar.iter_scans():
```

Comment:
do you always need all, or cutting down to ~3 or ~4 might be sufficient? (as we saw in practice)

Reply:
I don't understand the question. We have to poll the lidar fast enough that the buffer doesn't fill up or it starts to complain. I'm only keeping the one latest scan for now, since that's all that's needed for obstacle detection. We'll likely end up keeping additional scans, but I'll add that as we need it.